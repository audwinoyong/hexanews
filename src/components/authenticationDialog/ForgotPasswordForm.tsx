/**
 * Display forgot password form component.
 */

import * as React from 'react'
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator'
import Button from '@material-ui/core/Button'
import DialogActions from '@material-ui/core/DialogActions'
import DialogContent from '@material-ui/core/DialogContent'
import DialogContentText from '@material-ui/core/DialogContentText'
import DialogTitle from '@material-ui/core/DialogTitle'

import { forgetPasswordFormString } from '../../constants/string'
import styles from './styles'

export type Props = {
  handleCloseDialog: () => void
  onChangeAuthenticationState: () => void
}

export interface ComponentState {
  email: string
}

class ForgotPasswordForm extends React.Component<Props, ComponentState> {

  constructor(props: Props) {
    super(props)
    this.state = {
      email: '',
    }
  }

  disableSubmitButton() {
    const { email } = this.state
    return (email === '')
  }

  onForgotPassword = async () => {
    await null
  }

  public render() {
    const { email } = this.state
    const { handleCloseDialog, onChangeAuthenticationState } = this.props
    return (
      <div>
        <ValidatorForm
          ref='forgotPasswordForm'
          onSubmit={() => this.onForgotPassword()}
          instantValidate={false}
        >
          <DialogTitle>
            {forgetPasswordFormString.dialogTitle}
          </DialogTitle>
          <DialogContent>
            <DialogContentText style={styles.descriptionContainer}>
              {forgetPasswordFormString.dialogDescription}
            </DialogContentText>
            <TextValidator
              label='Email Address'
              onChange={(event: any) => this.setState({ email: event.target.value })}
              name='forgotPasswordEmail'
              style={styles.descriptionContainer}
              margin='dense'
              fullWidth
              type='email'
              value={email}
              helperText=' '
              validators={['required', 'isEmail']}
              errorMessages={['Please enter email', 'Please enter a valid email']}
            />
            <DialogContentText style={styles.footerContainer}>
              <span
                onClick={onChangeAuthenticationState}
                style={styles.footerLink}
              >
                {forgetPasswordFormString.backLogin}
              </span>
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button onClick={handleCloseDialog} color='primary'>
              {forgetPasswordFormString.cancelButton}
            </Button>
            <Button
              onClick={this.onForgotPassword}
              color='primary'
              disabled={this.disableSubmitButton()}
              type='submit'
            >
              {forgetPasswordFormString.submitButton}
            </Button>
          </DialogActions>
        </ValidatorForm>
      </div>
    )
  }
}

export default ForgotPasswordForm