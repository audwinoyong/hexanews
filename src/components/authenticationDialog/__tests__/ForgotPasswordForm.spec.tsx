/**
 * Unit tests for checking the behavior of Forgot Password Form components
 */

import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

import ForgotPasswordForm from '../ForgotPasswordForm'

Enzyme.configure({ adapter: new Adapter() })

describe('Forgot Password Form', () => {
  it('renders Forgot Password Form', () => {
    const component = renderer.create(
      <ForgotPasswordForm
        dispatch={jest.fn()}
        onChangeAuthenticationState={jest.fn()}
        isLoadingForgotPassword={false}
        forgotPasswordError=''
        handleOpenForgotPasswordSnackbar={jest.fn()}
        handleCloseDialog={jest.fn()}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})