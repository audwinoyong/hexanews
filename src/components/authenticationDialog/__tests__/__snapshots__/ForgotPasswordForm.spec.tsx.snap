// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password Form renders Forgot Password Form 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiDialogTitle-root-1"
    >
      <h2
        className="MuiTypography-root-2 MuiTypography-title-8"
      >
        Forgot Password
      </h2>
    </div>
    <div
      className="MuiDialogContent-root-27"
    >
      <p
        className="MuiTypography-root-2 MuiTypography-subheading-9 MuiDialogContentText-root-28"
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        Put your email here, we will send a confirmation to reset your password
      </p>
      <div
        aria-describedby={undefined}
        className="MuiFormControl-root-29 MuiFormControl-marginDense-31 MuiFormControl-fullWidth-32"
        style={
          Object {
            "marginBottom": 8,
          }
        }
      >
        <label
          className="MuiFormLabel-root-38 MuiInputLabel-root-33 MuiInputLabel-formControl-34 MuiInputLabel-animated-37 MuiInputLabel-marginDense-35"
          data-shrink={false}
          htmlFor={undefined}
        >
          Email Address
        </label>
        <div
          className="MuiInput-root-45 MuiInput-fullWidth-52 MuiInput-formControl-46 MuiInput-underline-49"
        >
          <input
            aria-invalid={false}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-53 MuiInput-inputType-56 MuiInput-inputMarginDense-54"
            defaultValue={undefined}
            disabled={false}
            id={undefined}
            name="forgotPasswordEmail"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={false}
            rows={undefined}
            type="email"
            value=""
          />
        </div>
        <p
          className="MuiFormHelperText-root-58 MuiFormHelperText-marginDense-61"
          id={undefined}
        >
           
        </p>
      </div>
      <p
        className="MuiTypography-root-2 MuiTypography-subheading-9 MuiDialogContentText-root-28"
        style={
          Object {
            "marginTop": 8,
          }
        }
      >
        <span
          onClick={[MockFunction]}
          style={
            Object {
              "color": "#03a87c",
              "cursor": "pointer",
            }
          }
        >
          Back to login
        </span>
      </p>
    </div>
    <div
      className="MuiDialogActions-root-65"
    >
      <button
        className="MuiButtonBase-root-93 MuiButton-root-67 MuiButton-text-69 MuiButton-textPrimary-70 MuiButton-flat-72 MuiButton-flatPrimary-73 MuiDialogActions-action-66"
        disabled={false}
        onBlur={[Function]}
        onClick={[MockFunction]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-68"
        >
          Cancel
        </span>
        <span
          className="MuiTouchRipple-root-96"
        />
      </button>
      <button
        className="MuiButtonBase-root-93 MuiButtonBase-disabled-94 MuiButton-root-67 MuiButton-text-69 MuiButton-textPrimary-70 MuiButton-flat-72 MuiButton-flatPrimary-73 MuiButton-disabled-87 MuiDialogActions-action-66"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="-1"
        type="submit"
      >
        <span
          className="MuiButton-label-68"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
</div>
`;
